pkg/reconciler/git/resolve.go

```diff
@@ -0,0 +1,104 @@
+package git
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/go-git/go-git/v5"
+
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/storage/memory"
+	"github.com/pkg/errors"
+)
+
+type refLister interface {
+	List(repoURL string) ([]*plumbing.Reference, error)
+}
+
+type remoteRefLister struct {
+}
+
+var defaultLister refLister = &remoteRefLister{}
```

Comment:
We shouldn't use a variable to change the behaviour of unit tests.  Also the encapsulation isn't very good.

Would recommend to bundle the functions in two different implementations:

      type revisionResolver struct {
         repository *git.Repository
         url string
      }

     func (r *revisionResolver) Resolve(rev string) {...code from resolveRevision ...}